compose/sample-apps_https.yml

```diff
@@ -0,0 +1,10 @@
+version: "3"
+services:
+  quarkus-test-agent:
```

Comment:
Now that the `vertx-fib-demo`s and `gameserver` sample applications that are deployed are also the new test-applications ones, and will contain a newer Agent version, I think they also need these same keystore configurations.

Reply:
Hmm. When I do `./smoketest.bash -t`, my `quarkus-test-agent` fails to register, with log output like this:

```
Caused by: java.util.concurrent.CompletionException: java.util.concurrent.CompletionException: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at io.cryostat.agent.CryostatClient.lambda$queryExistingCredentials$10(CryostatClient.java:219)
	... 10 more
Caused by: java.util.concurrent.CompletionException: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at io.cryostat.agent.CryostatClient.executeQuiet(CryostatClient.java:461)
	at io.cryostat.agent.CryostatClient.lambda$supply$28(CryostatClient.java:453)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 more
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:378)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:316)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1351)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1226)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1169)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:480)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:458)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:201)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1510)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1425)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426)
	at io.cryostat.agent.shaded.org.apache.shaded.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
	at io.cryostat.agent.shaded.org.apache.shaded.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at io.cryostat.agent.CryostatClient.executeQuiet(CryostatClient.java:459)
	... 8 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
	at java.base/sun.security.validator.Validator.validate(Validator.java:264)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:231)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:132)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1335)
	... 33 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:148)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:129)
	at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
```

`gameserver` fails like this, looks like this is probably because it hardcodes an `http` baseuri:

```
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: Caused by: java.util.concurrent.CompletionException: java.util.concurrent.CompletionException: io.cryostat.agent.shaded.org.apache.shaded.http.conn.HttpHostConnectException: Connect to auth:8080 [auth/10.89.61.162] failed: Connection refused
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.CryostatClient.lambda$queryExistingCredentials$10(CryostatClient.java:219)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	... 10 more
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: Caused by: java.util.concurrent.CompletionException: io.cryostat.agent.shaded.org.apache.shaded.http.conn.HttpHostConnectException: Connect to auth:8080 [auth/10.89.61.162] failed: Connection refused
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.CryostatClient.executeQuiet(CryostatClient.java:461)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.CryostatClient.lambda$supply$28(CryostatClient.java:453)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	... 6 more
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: Caused by: io.cryostat.agent.shaded.org.apache.shaded.http.conn.HttpHostConnectException: Connect to auth:8080 [auth/10.89.61.162] failed: Connection refused
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.CryostatClient.executeQuiet(CryostatClient.java:459)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	... 8 more
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: Caused by: java.net.ConnectException: Connection refused
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:542)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:592)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at java.base/java.net.Socket.connect(Socket.java:751)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:75)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	at io.cryostat.agent.shaded.org.apache.shaded.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
[15:56:59] [cryostat-agent-worker-1/INFO]: [STDERR]: 	... 19 more
```

The `vertx-fib-demo` instances are registering fine.

Reply:
I typically get the `sun.security.provider.certpath.SunCertPathBuilderException` when the `CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUST_ALL` and `CRYOSTAT_AGENT_WEBCLIENT_TLS_VERIFY_HOSTNAME` config names don't match the -agent version used for the quarkus-test-agent container image. I'll take a look right now at them

Reply:
So it's just the `:latest` images that are causing the `sun.security.provider.certpath.SunCertPathBuilderException` error. If you try the date or version tags, everything works as expected. I guess the `:latest` images being used are old?

Reply:
That's interesting... let me give it a try.

![image](https://github.com/user-attachments/assets/f2466a82-648a-4f2c-af55-948a82a104b4)

`latest`, `0.5.0-snapshot`, and `2024-07-15` are all supposed to be the same actual image. But maybe `podman` is using an older cached image with the same tag, even though I didn't pass the `-O` offline switch.

Reply:
Testing it again now with these latest rebuilt images, the `quarkus-cryostat-agent:latest` seems to be fine. The vertx and gameserver are still failing, but I think those are just because they are missing the required configuration to add the cert mount and/or set the baseuri to HTTPS.

Reply:
when I checked `podman images` they were all listed as the same images too.... so I'm not sure how they deploy differently

Reply:
I just deleted all my podman images and rebuilt everything and `:latest` images run okay now. Must have been an older cached image